src/gpu/intel/ocl/reusable_vectorized_lnorm.cl

```diff
@@ -32,80 +32,95 @@ VEC_SUM_DEFINE(float2)
 VEC_SUM_DEFINE(float4)
 VEC_SUM_DEFINE(float8)
 
+__attribute__((reqd_work_group_size(WG_SIZE, 1, 1)))
 __attribute__((intel_reqd_sub_group_size(SG_SIZE))) __kernel void
 lnorm_reusable_vectorized(__global SRC_DATA_T *src, __global float *mean,
         __global float *variance, dim_t reduce_size, __global DST_DATA_T *dst,
         __global WEI_DATA_T *scale, __global WEI_DATA_T *shift, float eps,
-        __global float *src_scale, __global float *dst_scale, int greads,
-        float rrs, dispatch_gws_rt_params_t gws_params) {
-    src = (GWS_GET_BUFFER_POS(SRC, gws_params, src)) - get_sub_group_local_id();
+        __global float *src_scale, __global float *dst_scale, float rrs,
+        dispatch_gws_rt_params_t gws_params) {
+    int sg_offset
+            = -(get_local_id(0)) + get_sub_group_id() * SG_SIZE * VECT_DT_N;
+    src = GWS_GET_BUFFER_POS(SRC, gws_params, src) + sg_offset;
+
+    mean = GWS_GET_BUFFER_POS(STAT, gws_params, mean);
+    variance = GWS_GET_BUFFER_POS(STAT, gws_params, variance);
+    FLT_ACC_DATA_T local_variance = CALCULATE_STATS ? 0.f : *variance;
+    FLT_ACC_DATA_T local_mean = CALCULATE_STATS ? 0.f : *mean;
+
+#if PVT_MEM_SIZE > 1
+    VECT_FLOAT_T val[PVT_MEM_SIZE];
+    unroll_for_by(N_UNROLL)(int sg_idx = 0, i = 0; i < PVT_MEM_SIZE;
+                            sg_idx += GROUP_STRIDE, i++) {
+        val[i] = CONVERT_VECT_FLOAT_T(AS_VECT_DATA_T(VECT_BLOCK_READ(
+                (const __global BLOCK_DATA_T *)(&src[sg_idx]))));
+    }
+#else
+    VECT_FLOAT_T val = CONVERT_VECT_FLOAT_T(AS_VECT_DATA_T(
+            VECT_BLOCK_READ((const __global BLOCK_DATA_T *)(src))));
+#endif
```

Comment:
I think the compiler should be able to optimize this incantation. Give it a shot and let me know.
```suggestion
    VECT_FLOAT_T val[PVT_MEM_SIZE];
    int sg_idx = 0;
    for (int i = 0; i < PVT_MEM_SIZE; i++) {
        val[i] = CONVERT_VECT_FLOAT_T(AS_VECT_DATA_T(VECT_BLOCK_READ(
                (const __global BLOCK_DATA_T *)(&src[sg_idx]))));
        sg_idx += GROUP_STRIDE;
    }
```